### PR TITLE
feat: format rasa core log format (#244)

### LINKED ISSUE #244: Rasa core logs should be properly formatted
I know, easier said than done...
What are our options to have logs in a usable format?

### PR DESCRIPTION
**Pull request by mathieubergeron**
## Description
The purpose of this PR is to:
- change formating of rasa core from plain text to JSON.
- add a `sender_id` attribute to every relevent log events, allowing us to better trace distinct conversations.

image

image

## Related issues
closes #244 

## Checklist

- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) guidelines 
- [x] All relevant PR sections are populated, irrelevant ones are removed

### CODE REVIEW

File: action-server/covidflow/setup_structlog.py [hunk 1]
```diff
@@ -9,6 +9,12 @@
 from pythonjsonlogger import jsonlogger
 from structlog.contextvars import merge_contextvars
 
+IS_STRUCTLOG_KEY = "is_structlog"
+
+###
+### FIXME: This file is a duplicate of "core/channels/setup_structlog.py".
+###
```
Comment (mathieubergeron) [inline action-server/covidflow/setup_structlog.py:16]: est-ce qu'on a un moyen facile de regler ça rapidement?

File: core/credentials.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 rest:
```
Comment (mathieubergeron) [inline core/credentials.yml:1]: not wrapping `rest` channel for now. This means that we do not have `sender_id` attributes in the logs for integration tests.

File: app.yml [hunk 3]
```diff
@@ -44,7 +44,7 @@ environment:
   ALEMBIC_DATABASE_URI: postgresql+psycopg2://$(SQL_TRACKER_STORE_USER):$(SQL_TRACKER_STORE_PASSWORD)@$(SQL_TRACKER_STORE_URL):5432/$(SQL_TRACKER_STORE_DB)
   DAILY_CHECKIN_URL_PATTERN: "https://covid19.dialogue.co/?lng={language}#/rasa/ci/{reminder_id}"
   REMINDER_ID_HASHIDS_MIN_LENGTH: 8
-  LOG_LEVEL: INFO
+  LOG_LEVEL: DEBUG
```
Comment (mathieubergeron) [inline app.yml:47]: log levels back to DEBUG since they now are properly formated

File: core/credentials.yml (hunk 2 above)
Comment (mathieubergeron) [reply core/credentials.yml:1]: Actually, since it is really easy, i'll do it now.

File: action-server/covidflow/setup_structlog.py (hunk 1 above)
Comment (fparga) [reply action-server/covidflow/setup_structlog.py:16]: On peut avoir des dependances locales (avec un path) avec poetry mais pas sur de ce que ca donnerait apres dans Docker

File: core/credentials.yml (hunk 2 above)
Comment (mathieubergeron) [reply core/credentials.yml]: (Done)

File: action-server/covidflow/setup_structlog.py [hunk 4]
```diff
@@ -84,6 +97,14 @@ def _add_record(log_record, key, value):
 class CustomJsonFormatter(jsonlogger.JsonFormatter):
     def add_fields(self, log_record, record, message_dict):
         super(CustomJsonFormatter, self).add_fields(log_record, record, message_dict)
+
+        if log_record.get(IS_STRUCTLOG_KEY) is True:
+            del log_record[IS_STRUCTLOG_KEY]
+        else:
+            # Applying contextvars to non-structlog events
+            contextvars = merge_contextvars(None, None, {})
```
Comment (fparga) [inline action-server/covidflow/setup_structlog.py:105]: Not sure to understand what this is doing

**Comment by fparga** [comment]
Don't forget to add a PR description

File: action-server/covidflow/setup_structlog.py (hunk 4 above)
Comment (mathieubergeron) [reply action-server/covidflow/setup_structlog.py:105]: `merge_contextvars` is a structlog processor that basically merges the contextvars with the third argument here. I'm reusing this to also append contextvars binded attributes to the non-structlog log event.

We are binding such attribute here: https://github.com/dialoguemd/covidflow/pull/261/files#diff-25f4a67a16aaa483d26997883709da28R19

File: core/channels/setup_structlog.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+
+def _add_structlog_indicator(logger, name, log_event):
+    log_event["extra"][IS_STRUCTLOG_KEY] = True
+    return log_event
+
+
+def _add_record(log_record, key, value):
+    if not log_record.get(key):
+        log_record[key] = value
```
[the remark this answers is not in the document]
Comment (mathieubergeron) [reply core/channels/setup_structlog.py:94]: i've added what i could

**Pull request merged**